zerver/lib/users.py

```diff
@@ -524,6 +653,76 @@ def get_cross_realm_dicts() -> List[Dict[str, Any]]:
     return result
 
 
+def get_data_for_inaccessible_user(
+    user_id: int, user_is_active: bool, realm: Realm
+) -> Dict[str, Any]:
+    user_email = Address(username=f"user{user_id}", domain=get_fake_email_domain(realm)).addr_spec
+
+    # We just set date_joined field to UNIX epoch.
+    user_date_joined = timestamp_to_datetime(0)
+
+    user_dict = dict(
+        email=user_email,
+        user_id=user_id,
+        avatar_version=1,
+        is_admin=False,
+        is_owner=False,
+        is_guest=False,
+        is_billing_admin=False,
+        role=UserProfile.ROLE_MEMBER,
+        is_bot=False,
+        full_name="Unknown user",
```

Comment:
We should probably tag "Unknown user" for translation, right?

Reply:
Done this. Also added a constant which is set to "Unknown user".